gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/SpecStoreChangeMonitor.java

```diff
@@ -171,5 +178,6 @@ protected void createMetrics() {
     this.deletedSpecs = this.getMetricContext().contextAwareMeter(RuntimeMetrics.GOBBLIN_SPEC_STORE_MONITOR_DELETED_SPECS);
     this.unexpectedErrors = this.getMetricContext().contextAwareMeter(RuntimeMetrics.GOBBLIN_SPEC_STORE_MONITOR_UNEXPECTED_ERRORS);
     this.messageProcessedMeter = this.getMetricContext().contextAwareMeter(RuntimeMetrics.GOBBLIN_SPEC_STORE_MESSAGE_PROCESSED);
+    this.produceToConsumeLag = this.getMetricContext().newContextAwareGauge(RuntimeMetrics.GOBBLIN_SPEC_STORE_PRODUCE_TO_CONSUME_LAG, () -> produceToConsumeLagValue);
```

Comment:
clearly some impl overlap between these two change monitor classes.  may be worth investigating later whether they could both use the same base class to take care of the message reading, caching, and also monitoring.  since they both read `GenericChangeStoreEvent`s), seems quite promising. only store-specific logic need be in either of the two derived classes.

Reply:
Yea I agree, there's potential to refactor these two monitor classes by creating a base case with the repeated code then handle the store specific logic. I'll keep that in mind for future improvements.